osu.Game/Overlays/Music/Playlist.cs

```diff
@@ -29,7 +30,7 @@ public void Filter(FilterCriteria criteria)
             var items = (SearchContainer<RearrangeableListItem<BeatmapSetInfo>>)ListContainer;
 
             foreach (var item in items.OfType<PlaylistItem>())
-                item.InSelectedCollection = criteria.Collection?.Beatmaps.Any(b => b.BeatmapSet.Equals(item.Model)) ?? true;
+                item.InSelectedCollection = criteria.Collection?.Beatmaps.Any(b => b.MatchesOnlineID(item.Model)) ?? true;
```

Comment:
This breaks filtering for unsubmitted beatmaps. Reproduction steps:

1. Create new collection
2. Add local unsubmitted map without a local ID to collection
3. Open playlist overlay
4. Filter by collection created in step (1)
5. Filtering results in an empty list

Reply:
Ah, this is an oversight. I thought this was the multiplayer playlist class. I think we probably should change the naming of one of the two - when working through these refactors I've made this mistake multiple times (they can easily get intertwined).

Also seems like this may need the equality you propose so I'll look at adding that back in this pull if required.

Reply:
Let's also pretend this wasn't comparing a `Beatmap` to a `BeatmapSet`...

I think I'm going to make `MatchesOnlineID` per type in this PR.